prometheus-to-sd/config/dynamic_source.go

```diff
@@ -72,32 +88,65 @@ func createOptionsForPodSelection(nodeName string, sources map[string]url.URL) v
 		nameList += key + ","
 	}
 	labelSelector := fmt.Sprintf("%s in (%s)", nameLabel, nameList)
-	return v1.ListOptions{
+	listOptions := v1.ListOptions{
```

Comment:
What happens if a customer creates their own pod with k8s-app label, for example "k8s-app: heapster"? You don't rely on anything that identifies pod unambiguously (like pod namespace + pod name, or just pod id), therefore you can get multiple events related to different pods, but AFAIU your code will think it's the same pod.

Note that this is a realistic scenario, for example some customers may disable stackdriver monitoring and deploy their own heapster using deployments from here: https://github.com/kubernetes/heapster/tree/master/deploy/kube-config.

Reply:
I am using pod namespace, which is kube-system, I am not sure that we can rely on pod name or pod id though